File: java/LongestIncreasingSubsequence.java
// Runtime: 49ms
// Memory Usage: 44.7MB

class Solution{
    public int lengthOfLIS(int[] nums) {
        int[] dpArr = new int[nums.length];
        Arrays.fill(dpArr, 1);

        for(int i = 1; i < nums.length; i++) {

            for(int j = 0; j < i; j++) {

                if(nums[j] < nums[i]) {

                    dpArr[i] = Math.max(
                        dpArr[i], 
                        dpArr[j] + 1
                    );
                }
            }
        }
        return Arrays.stream(dpArr).max().getAsInt();
}
    
}

File: python/LongestIncreasingSubsequence.py
# Runtime: 1847ms
# MemoryUsage: 16.8MB

class Solution:
    def lengthOfLIS(self, nums: List[int]) -> int:
        result = [1] * len(nums)
        for right in range(1, len(nums)):
            num_right = nums[right]
            for left in range(0, right):
                num_left = nums[left]
                if num_right > num_left:
                    result[right] = max(
                        result[right], 
                        result[left]+1
                    )
        return max(result)
